test/signup.test.js

```diff
@@ -0,0 +1,168 @@
+const request = require('supertest'),
+  app = require('../app.js'),
+  Users = require('../app/models').user,
+  { factory } = require('factory-girl'),
+  validationErrorStatus = 401,
+  createdCorrectlyStatus = 200,
+  firstName = 'fn',
+  lastName = 'ln',
+  correctPassword = 'password',
+  correctEmail = 'email@wolox.com.ar';
+
+describe('Test invalid password', () => {
+  test('Create user with short password', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        last_name: lastName,
+        email: correctEmail,
+        password: 'pass'
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user with password with invalid characters', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        last_name: lastName,
+        email: correctEmail,
+        password: 'passwo..rd'
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+});
+
+describe('Test invalid email', () => {
+  test('Create user with email with domain different from wolox', () => {
+    expect.assertions(1);
+    return request(app)
+      .post('/users')
+      .send({
+        first_name: firstName,
+        last_name: lastName,
+        email: 'email@email.com.ar',
+        password: correctPassword
+      })
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+
+  test('Create user with used email', () => {
+    factory.define('user', Users, buildOptions => ({
+      first_name: buildOptions.firstName,
+      last_name: buildOptions.lastName,
+      email: buildOptions.email,
+      password: buildOptions.password
+    }));
+    expect.assertions(1);
+    return factory
+      .create('user', { firstName, lastName, email: correctEmail, password: correctPassword })
+      .then(() =>
+        request(app)
+          .post('/users')
+          .send({
+            first_name: firstName,
+            last_name: lastName,
+            email: correctEmail,
+            password: correctPassword
+          })
+      )
+      .then(response => {
+        expect(response.status).toBe(validationErrorStatus);
+      });
+  });
+});
+
+describe('Test missing parameters', () => {
```

Comment:
Can't you use a describe.each here? the only thing that changes between each test in this describe is the params you send.